From 258d36c4df2a094af1a891eb6ecb3d9bcdfe06eb Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Mon, 6 Apr 2026 06:58:29 +0000
Subject: [PATCH] powerpc/powernv: Add a null pointer check in
 opal_event_init()

jira VULN-35456
cve CVE-2023-52686
commit-author Kunwu Chan <chentao@kylinos.cn>
commit 8649829a1dd25199bbf557b2621cedb4bf9b3050

kasprintf() returns a pointer to dynamically allocated memory
which can be NULL upon failure.

Fixes: 2717a33d6074 ("powerpc/opal-irqchip: Use interrupt names if present")
	Signed-off-by: Kunwu Chan <chentao@kylinos.cn>
	Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
Link: https://msgid.link/20231127030755.1546750-1-chentao@kylinos.cn
(cherry picked from commit 8649829a1dd25199bbf557b2621cedb4bf9b3050)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
---
 arch/powerpc/platforms/powernv/opal-irqchip.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/powerpc/platforms/powernv/opal-irqchip.c b/arch/powerpc/platforms/powernv/opal-irqchip.c
index c164419e254d..dcec0f760c8f 100644
--- a/arch/powerpc/platforms/powernv/opal-irqchip.c
+++ b/arch/powerpc/platforms/powernv/opal-irqchip.c
@@ -278,6 +278,8 @@ int __init opal_event_init(void)
 		else
 			name = kasprintf(GFP_KERNEL, "opal");
 
+		if (!name)
+			continue;
 		/* Install interrupt handler */
 		rc = request_irq(r->start, opal_interrupt, r->flags & IRQD_TRIGGER_MASK,
 				 name, NULL);
-- 
2.47.3

